test/mongoose_subdomain_core_SUITE.erl

```diff
@@ -0,0 +1,567 @@
+    mongoose_domain_core:delete("domain.test"),
+    mongoose_subdomain_core:sync(),
+    ?assertEqual({ok, ExpectedSubdomainInfo1},
+                 mongoose_subdomain_core:get_subdomain_info(<<"sub.domain.test">>)),
+    %%----------------------------------------------------------------
+    %% testing type #2 subdomain names collision + double registration
+    %%----------------------------------------------------------------
+    ?assertEqual(ok, mongoose_subdomain_core:register_subdomain(?DYNAMIC_HOST_TYPE1,
+                                                                Pattern3, Handler)),
+    ?assertEqual({error, already_registered},
+                 mongoose_subdomain_core:register_subdomain(?DYNAMIC_HOST_TYPE1,
+                                                            Pattern3, Handler)),
+    ?assertEqual({error, subdomain_already_exists},
+                 mongoose_subdomain_core:register_subdomain(?DYNAMIC_HOST_TYPE2,
+                                                            Pattern3, Handler)),
+
+    ?assertEqual(4, get_subdomains_table_size()),
+    ExpectedSubdomainInfo2 =
+        #{host_type => ?DYNAMIC_HOST_TYPE1, subdomain_pattern => Pattern3,
+          parent_domain => undefined, packet_handler => Handler,
+          subdomain => <<"sub.domain.fqdn">>},
+    ?assertEqual({ok,ExpectedSubdomainInfo2},
+                 mongoose_subdomain_core:get_subdomain_info(<<"sub.domain.fqdn">>)),
+    %%------------------------------------------
+    %% testing type #3 subdomain names collision
+    %%------------------------------------------
+    mongoose_domain_core:insert(<<"fqdn">>, ?DYNAMIC_HOST_TYPE1, dummy_src),
+    mongoose_subdomain_core:sync(),
+    ?assertEqual(5, get_subdomains_table_size()),
+    ExpectedSubdomainInfo3 =
+        #{host_type => ?DYNAMIC_HOST_TYPE1, subdomain_pattern => Pattern3,
+          parent_domain => undefined, packet_handler => Handler,
+          subdomain => <<"sub.domain.fqdn">>},
+    ?assertEqual({ok, ExpectedSubdomainInfo3},
+                 mongoose_subdomain_core:get_subdomain_info(<<"sub.domain.fqdn">>)),
+    %%---------------------------------------
+    %% check that all collisions are detected
+    %%---------------------------------------
+    ?assertEqual([[ExpectedSubdomainInfo1,
+                   ExpectedSubdomainInfo1#{subdomain_pattern := Pattern1,
+                                           parent_domain => <<"domain.test">>}],
+                  [ExpectedSubdomainInfo2,
+                   ExpectedSubdomainInfo2#{host_type := ?DYNAMIC_HOST_TYPE2}],
+                  [ExpectedSubdomainInfo3,
+                   ExpectedSubdomainInfo3#{subdomain_pattern := Pattern2,
+                                           parent_domain := <<"fqdn">>}]],
+                 get_list_of_subdomain_collisions()),
+    no_domain_collisions().
+
+detects_domain_subdomain_collisions(_Config) ->
+    %% there are two possible domain/subdomain names collisions:
```

Comment:
Could you reformat the sentences here as well? Just one rule: when a sentence ends with ".", start it with a capital letter.